packages/framework/ai-collab/src/aiCollabApi.ts

```diff
@@ -0,0 +1,145 @@
+// eslint-disable-next-line import/no-named-as-default
+import type OpenAI from "openai";
+
+/**
+ * OpenAI client options for the {@link AiCollabOptions} interface.
+ *
+ * @alpha
+ */
+export interface OpenAiClientOptions {
+	/**
+	 * The {@link OpenAI} client to use for the AI collaboration.
+	 */
+	client: OpenAI;
+	/**
+	 * The name of the target OpenAI model to use for the AI collaboration.
+	 */
+	modelName?: string;
+}
+
+/**
+ * Options for the AI collaboration.
+ *
+ * @alpha
+ */
+export interface AiCollabOptions<TSchema extends ImplicitFieldSchema> {
+	openAI: OpenAiClientOptions;
+	/**
+	 * The view of your Shared Tree.
+	 * @remarks Its is recommended to pass a branch of your current tree view so the AI has a separate canvas to work on
+	 * and merge said branch back to the main tree when the AI is done and the user accepts
+	 */
+	treeView: TreeView<TSchema>;
+	/**
+	 * The specific tree node you want the AI to collaborate on. Pass the root node of your tree if you intend
+	 * for the AI to work on the entire tree.
+	 * @remarks
+	 * - Optional root nodes are not supported
+	 * - Primitive root nodes are not supported
+	 */
+	treeNode: TreeNode;
+	prompt: {
+		/**
+		 * The context to give the LLM about its role in the collaboration.
+		 */
+		systemRoleContext: string;
+		/**
+		 * The request from the users to the LLM.
+		 */
+		userAsk: string;
+	};
+	/**
+	 * Limiters are various optional ways to limit this library's usage of the LLM.
+	 */
+	limiters?: {
+		abortController?: AbortController;
+		/**
+		 * The maximum number of sequential errors the LLM can make before aborting the collaboration.
+		 */
+		maxSequentialErrors?: number;
+		/**
+		 * The maximum number of model calls the LLM can make before aborting the collaboration.
+		 */
+		maxModelCalls?: number;
+		/**
+		 * The maximum token usage limits for the LLM.
+		 */
+		tokenLimits?: TokenUsage;
+	};
+	/**
+	 * When enabled, the LLM will be prompted to first produce a plan based on the user's ask before generating changes to your applications data
+	 */
+	planningStep?: boolean;
+	/**
+	 * When enabled, the LLM will be prompted with a final review of the changes they made to confimr their validity.
+	 */
+	finalReviewStep?: boolean;
+	/**
+	 * An optional validator function that can be used to validate the new content produced by the LLM.
+	 */
+	validator?: (newContent: TreeNode) => void;
+	/**
+	 * When enabled, the library will console.log information useful for debugging the AI collaboration.
+	 */
+	dumpDebugLog?: boolean;
+}
+
+/**
+ * A successful response from the AI collaboration.
+ *
+ * @alpha
+ */
+export interface AiCollabSuccessResponse {
+	/**
+	 * The status of the Ai Collaboration.
+	 * A 'success' status indicates that the AI collaboration was successful at creating changes.
+	 */
+	status: "success";
+	/**
+	 * {@inheritDoc TokenUsage}
+	 */
+	tokenUsage: TokenUsage;
```

Comment:
Minor: I'd consider calling this field `tokensUsed` to clarify that it's in the past, i.e. these tokens have already been used, in contrast with tokens we plan/estimate to use, tokens you have remaining, or a maximum number of tokens we will allow using. The shared `TokenUsage` type makes sense, I just think the field name could create ambiguity.

Reply:
I've been going back and forth on how to distinguish between token limits and token usage because if you create two distinct types, they are nearly the same. However, I decided to go with creating two unique types, `TokenUsage` and `TokenLimits` (TokenLimits make the input and output tokens optional) as of commit f7425c23bc783cf57d9fdbc2ce059ce131026016. I've also updated the field name here to `tokensUsed` per your suggestion.

I also added a